luigi/contrib/redshift.py

```diff
@@ -52,7 +138,7 @@ class RedshiftTarget(postgres.PostgresTarget):
     use_db_timestamps = False
 
 
-class S3CopyToTable(rdbms.CopyToTable):
+class S3CopyToTable(rdbms.CopyToTable, _CredentialsMixin):
```

Comment:
Is this the correct order of inheritance? Based on this http://stackoverflow.com/questions/825945/abstract-class-mixin-multiple-inheritance-in-python it should be ordered the other way around. I couldn't test it yet.

Reply:
Where did we land on this thread? Do your tests cover this (potential) issue?

Reply:
This is what my most recent commit is for. I test loading the credentials from the different available sources. Let me know if you want to see other tests.

Reply:
This looks good. Just to clarify: What I missed earlier is the fact that within this PR the `aws` credentials are no longer abstract. Potentially missing credentials are now handled explicitly in the mixin. Therefore order of inheritance is indeed irrelevant here. 👍